Robust_Lab.qmd

```diff
@@ -12,93 +13,368 @@ execute:
+      annotate("text", x=vline[1]-0.05, y=50,
+               color = "#D81B60",
+               label="95% CI", angle=90) +
+      annotate("text", x=vline[3]-0.05, y=100,
+               color = "#004D40",
+               label="BCA CI", angle=90)
+    ```
+
     d\. For the median bootstraps, plot the 95% confidence intervals (Percentile and BCa). Use `geom_vline and annotate` to mark the lines noting what they represent.
 
 3.  You want to test whether the following paired samples are significantly different from one another: pre = \[22,25,17,24,16,29,20,23,19,20\], post = \[18,21,16,22,19,24,17,21,23,18\]. Often researchers would run a paired sampled t-test, but you are concerned the data does not follow a normal distribution.
 
-    a.  Calculate the paired differences, that is post - pre, which will result in a vector of paired differences (pdiff0 = post - pre)
+    a\. Calculate the paired differences, that is post - pre, which will result in a vector of paired differences (paired_diff = post - pre)
+
+    ```{r}
+
+    # input data
+    pre <- c(22,25,17,24,16,29,20,23,19,20)
+    post <- c(18,21,16,22,19,24,17,21,23,18)
+    # calculate change
+    paired_diff <- post - pre
+
+    ```
+
+    b\. Calculate the mean of the paired differences (X_paired_diff)
+
+    ```{r}
+
+    # calculate mean
+    mean_paired_diff <- mean(paired_diff)
+
+    ```
 
-    b\. Calculate the mean of the paired differences (Xpdiff0)
+    c\. Bootstrap b) with replacement (pdiff1) and plot the histogram with `ggplot2`.
 
-    d\. Bootstrap b) with replacement (pdiff1) and plot the histogram with `ggplot2`.
+    ```{r}
 
-    e\. Calculate the 95% confidence intervals (BCa). What can you infer from this?
+    # bootstrap mean differences
+    paired_diff_boot <- one.boot(paired_diff, mean, R=1000)
+    hist(paired_diff_boot)
 
-    f\. Plot bootstrap mean along with 95% CIs (with `ggplot2`). Use annotate to note what the vertical lines represent.
+    ```
 
-4.  Pepper Joe measured the length and heat of 85 chili peppers. He wants to know if smaller peppers are hotter than longer peppers.
+    d\. Calculate the 95% confidence intervals (BCa). What can you infer from this?
+
+    ```{r}
+
+    # calculate 95% CI using BCA
+    boot.ci(paired_diff_boot, type = "bca", R=1000)
+    ```
+
+    > The 95% confidence interval ranges from -3.1 to 0.4, which means that the "true" change could be zero.
+
+    e\. Plot bootstrap mean along with 95% CIs (with `ggplot2`). Use annotate to note what the vertical lines represent.
+
+4.  Pepper Joe measured the length and heat of 85 chilli peppers. He wants to know if smaller peppers are hotter than longer peppers.
 
     ```{r}
     #read data.table to read in
-    chili<- read.delim("/Users/jasongeller/Documents/03-Robust_Methods/data/chillis.csv")
+    chilli <- read.delim("https://raw.githubusercontent.com/jgeller112/psy504-advanced-stats/main/_site/slides/03-Robust_Methods/data/chillis.csv")
+    ```
```

Comment:
Sorry. I was not clear here. My fault. This is the activity from class notes that has you performing a bootstrapped regression. Below is how you would do it: 

boot.function <- function(data, indices) {
resampled_data = data[indices,]
model<-glm(HEAT~LENGTH, data=resampled_data)
return(coefficients(model))}

results <- boot(chili, boot.function, R=10000)

boot.ci(results, type="bca", R=10000)